error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/ExplicitArgumentEnumeration.java

```diff
@@ -0,0 +1,200 @@
+                  ImmutableSet.class.getCanonicalName(),
+                  List.class.getCanonicalName(),
+                  Set.class.getCanonicalName())
+              .named("of"),
+          staticMethod().onClass(Arrays.class.getCanonicalName()).named("asList"));
+  private static final Matcher<ExpressionTree> IMMUTABLE_COLLECTION_BUILDER =
+      instanceMethod().onDescendantOf(ImmutableCollection.Builder.class.getCanonicalName());
+  private static final Matcher<ExpressionTree> OBJECT_ENUMERABLE_ASSERT =
+      instanceMethod().onDescendantOf("org.assertj.core.api.ObjectEnumerableAssert");
+  private static final Matcher<ExpressionTree> STEP_VERIFIER_STEP =
+      instanceMethod().onDescendantOf("reactor.test.StepVerifier.Step");
+  private static final ImmutableTable<Matcher<ExpressionTree>, String, String> ALTERNATIVE_METHODS =
+      ImmutableTable.<Matcher<ExpressionTree>, String, String>builder()
+          .put(IMMUTABLE_COLLECTION_BUILDER, "addAll", "add")
+          .put(OBJECT_ENUMERABLE_ASSERT, "containsAnyElementsOf", "containsAnyOf")
+          .put(OBJECT_ENUMERABLE_ASSERT, "containsAll", "contains")
+          .put(OBJECT_ENUMERABLE_ASSERT, "containsExactlyElementsOf", "containsExactly")
+          .put(
+              OBJECT_ENUMERABLE_ASSERT,
+              "containsExactlyInAnyOrderElementsOf",
+              "containsExactlyInAnyOrder")
+          .put(OBJECT_ENUMERABLE_ASSERT, "doesNotContainAnyElementsOf", "doesNotContain")
+          .put(OBJECT_ENUMERABLE_ASSERT, "hasSameElementsAs", "containsOnly")
+          .put(STEP_VERIFIER_STEP, "expectNextSequence", "expectNext")
+          .build();
+
+  /** Instantiates a new {@link ExplicitArgumentEnumeration} instance. */
+  public ExplicitArgumentEnumeration() {}
+
+  @Override
+  public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState state) {
+    if (tree.getArguments().size() != 1) {
+      return Description.NO_MATCH;
+    }
+
+    MethodSymbol method = ASTHelpers.getSymbol(tree);
+    if (!isUnaryIterableAcceptingMethod(method, state)) {
+      return Description.NO_MATCH;
+    }
+
+    ExpressionTree argument = tree.getArguments().get(0);
+    if (!(argument instanceof MethodInvocationTree)
+        || !EXPLICIT_ITERABLE_CREATOR.matches(argument, state)) {
+      return Description.NO_MATCH;
+    }
+
+    return trySuggestCallingVarargsOverload(method, (MethodInvocationTree) argument, state)
+        .or(() -> trySuggestCallingCustomAlternative(tree, (MethodInvocationTree) argument, state))
+        .map(fix -> describeMatch(tree, fix))
+        .orElse(Description.NO_MATCH);
+  }
+
+  private static boolean isUnaryIterableAcceptingMethod(MethodSymbol method, VisitorState state) {
+    List<VarSymbol> params = method.params();
+    return !method.isVarArgs()
+        && params.size() == 1
+        && ASTHelpers.isSubtype(params.get(0).type, state.getSymtab().iterableType, state);
+  }
```

Comment:
Pitest flags that the `params.size() == 1` can be mutated away. This is true thanks to the initial `tree.getArguments().size() != 1` check, but it doesn't seem right to drop this condition; it makes the code less clear, IMHO.